research-software-directory/RSD-as-a-service
# feat: migrate user documentation to main repo

# Migrate user documentation to main repo

**This is the first attempt. I do expect some feedback at least from @jmaassen and @cmeessen** :-)

Closes #911

### Changes proposed in this pull request:
* Move the content of users documentation from documentation repo to main RSD repo into the folder documentation
* Setup Docusaurus in the `\documentation` folder
* Added `maintaining organisation` page contributed by Hellmholtz. The content is slightly adjusted and the images are updated to v2
* (Proposal) Created 4 sections in the documentation:
    * User manual: documentation for RSD users and maintainers
    * API services: postgREST api and OAI-PMH description
    * RSD administration: section about running RSD instance and RSD administration pages
    * Contribute to RSD: this section is for contributors and contains information relevant for the contributors
* Add `documentation` service to main docker-compose.yml and to docker-compose.yml in the deployment folder. 
* Extend release action to create the documentation image using the same version as the rest of RSD services

### To do
- `Last changed date` info at the right bottom corner of the page is not working when building documentation as a  Docker image because Git information is not available during the build. If this feature is desired we would need to further investigate how to include git information without copying complete RSD project into Docker image.

### How to test:
* `make start` to build all services locally
* navigate to home page and start checking documentation links:
   * "Sing up to contribute" button has documentation link at the top of the modal
   * "Register your organisation" button has documentation link at the top of the modal
   * "Demo" button should open documentation at demo page
   * "Docs" button should open intro page of documentation
   * "FAQ" button should open FAQ section of user documentation
* check the documentation links in the footer section of RSD home page
* documentation app: view README.md file in the root of the documentation folder. It describes how to work with the current setup. Try running documentation locally and making changes.

## Example
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/ff99ed08-d6ee-4c09-b3a3-a57b0a3e419c)

PR Checklist:
